workers/go/throughputstress/workflow.go

```diff
@@ -20,6 +19,110 @@ const (
 
 var activityStub = Activities{}
 
+// ThroughputStressExecutorWorkflow drives the throughputstress scenario. It exists so that it's
+// trivial to start the scenario by simply invoking this workflow, without needing to use an omes
+// binary. It also provides durability for very long-running configurations.
+func ThroughputStressExecutorWorkflow(
+	ctx workflow.Context,
+	params *throughputstress.ExecutorWorkflowInput,
+) error {
+	// Make sure the search attribute is registered
+	actCtx := workflow.WithActivityOptions(ctx, defaultActivityOpts())
+	err := workflow.ExecuteActivity(
+		actCtx,
+		activityStub.EnsureSearchAttributeRegistered,
+		workflow.GetInfo(ctx).Namespace,
+	).Get(ctx, nil)
+	if err != nil {
+		return err
+	}
+	// Complain if there are already existing workflows with the provided run id
+	var count int64
+	err = workflow.ExecuteActivity(
+		actCtx,
+		activityStub.VisibilityCount,
+		fmt.Sprintf("%s='%s'", throughputstress.ThroughputStressScenarioIdSearchAttribute, params.RunID),
+	).Get(ctx, &count)
+	if count > 0 {
+		return fmt.Errorf("there are already %d workflows with scenario Run ID '%s'", count, params.RunID)
+	}
+
+	// Run the load
+	timeout := time.Duration(1*params.LoadParams.Iterations) * time.Minute
+
+	startingChan := workflow.NewChannel(ctx)
+	allWorkflowsDone, allWorkflowsDoneSet := workflow.NewFuture(ctx)
+	completedTopLevelWorkflowCount := 0
+	completedWorkflowCount := 0
+
+	for i := 0; i < params.MaxConcurrent; i++ {
+		workflow.Go(ctx, func(ctx workflow.Context) {
+			for {
+				var iterNumber int
+				more := startingChan.Receive(ctx, &iterNumber)
+				if !more {
+					return
+				}
+				wfID := fmt.Sprintf("throughputStress-%s-%d", params.RunID, iterNumber)
+				childCtx := workflow.WithChildOptions(ctx, workflow.ChildWorkflowOptions{
+					WorkflowID:               wfID,
+					WorkflowExecutionTimeout: timeout,
+					SearchAttributes: map[string]interface{}{
+						throughputstress.ThroughputStressScenarioIdSearchAttribute: params.RunID,
+					},
+				})
+				var result throughputstress.WorkflowOutput
+				err := workflow.ExecuteChildWorkflow(
+					childCtx,
+					ThroughputStressWorkflow,
+					params.LoadParams,
+				).Get(ctx, &result)
+				if err != nil {
+					allWorkflowsDoneSet.SetError(err)
+				}
+				completedTopLevelWorkflowCount++
+				completedWorkflowCount += 1 + result.ChildrenSpawned + result.TimesContinued
+				if completedTopLevelWorkflowCount == params.Iterations {
+					allWorkflowsDoneSet.SetValue(nil)
+				}
+			}
+		})
+	}
+
+	workflow.Go(ctx, func(ctx workflow.Context) {
+		for i := 0; i < params.Iterations; i++ {
```

Comment:
If iterations are greater than some threshold you'll want to split this into multiple child workflows each executing some of the workflows. For a very large number of iterations, you might want a multi-level tree structure to fanout.

Reply:
I'll add a comment about this technique so it can be added if/when this gets used that way. Not immediately clear it will be.